app/mqtt_reader.py

```diff
@@ -21,19 +22,19 @@ def _decode_message(message):
 class MqttReader:
     def __init__(
         self,
-        host: str,
-        port: int,
-        topics: list[str],
+        configuration: Configuration,
```

Comment:
I like this change

Reply:
yes it's consistent with the way the other client objects are inited, and gives them all responsibility for getting whatever they need from the configuration